lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -453,6 +453,18 @@ def _extract_collection_from_git(repo_url, coll_ver, b_path):
             proc_err,
         )
 
+    if pathlib.Path(str(b_checkout_path) + "/.gitmodules").is_file():
+        git_submodule_cmd = git_executable, 'submodule', 'update', '--init', '--recursive'
+        try:
+            subprocess.check_call(git_submodule_cmd, cwd=b_checkout_path)
+        except subprocess.CalledProcessError as proc_err:
+            raise_from(
+                AnsibleError(
+                    'Failed to download submodules'
+                ),
+                proc_err,
+            )
```

Comment:
Controller-side code is already py3-only so there shouldn't be any need for the six helper
```suggestion
            raise AnsibleError('Failed to download submodules') from proc_err

```